src/Microsoft.AspNetCore.Mvc.Core/Internal/FileResultExecutorBase.cs

```diff
@@ -2,27 +2,164 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Internal;
 using Microsoft.Extensions.Logging;
 using Microsoft.Net.Http.Headers;
 
 namespace Microsoft.AspNetCore.Mvc.Internal
 {
     public class FileResultExecutorBase
     {
+        private PreconditionState _ifMatchState;
+        private PreconditionState _ifNoneMatchState;
+        private PreconditionState _ifModifiedSinceState;
+        private PreconditionState _ifUnmodifiedSinceState;
+
         public FileResultExecutorBase(ILogger logger)
         {
             Logger = logger;
+            _ifMatchState = PreconditionState.Unspecified;
+            _ifNoneMatchState = PreconditionState.Unspecified;
+            _ifModifiedSinceState = PreconditionState.Unspecified;
+            _ifUnmodifiedSinceState = PreconditionState.Unspecified;
+        }
+
+        public enum PreconditionState
+        {
+            Unspecified,
+            NotModified,
+            ShouldProcess,
+            PreconditionFailed,
         }
 
         protected ILogger Logger { get; }
 
+        public PreconditionState GetPreconditionState()
```

Comment:
I don't see this being used anywhere, so you can remove this, the public enum above and the GetMaxPreconditionState method below